src/main/java/com/tabnine/general/Utils.java

```diff
@@ -126,4 +131,30 @@ public static boolean isUnitTestMode() {
     return ApplicationManager.getApplication() == null
         || ApplicationManager.getApplication().isUnitTestMode();
   }
+
+  public static String trimEndSlashAndWhitespace(String text) {
+    if (text == null) {
+      return null;
+    }
+    return text.replaceAll("/\\s*$", "");
+  }
+
+  public static void SetCustomRepository(String cloud2Url) {
```

Comment:
1. this is not c# here lol
2. rename argument
```suggestion
  public static void setCustomRepository(String url) {
```